server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxChangesMethodContract.scala

```diff
@@ -1023,6 +1023,70 @@ trait MailboxChangesMethodContract {
       }
     }
 
+    @Test
+    def mailboxChangesShouldReturnUpdatedChangesWhenSubscribedOnlyPerUser(server: GuiceJamesServer): Unit = {
```

Comment:
I've been hesitating a lot on how to test it... Is it the right test and how can we be sure there is no change for the other sharee actually? Thread.sleep() given a certain time then check Andre is still unsubscribed? Doesn't really feel precise...

Should we have similar tests in Mailbox/Set?

Reply:
(and yes this test is incomplete and I miss unsubscribe case)

Reply:
maybe use a subscription probe to check?

Reply:
Yes that's not the issue... more like is it acceptable to say I wait a few seconds to see that the subscription doe snot change for the sharee? I don't see an other way though :)

Reply:
Correct me if I am wrong because I read quickly.

> that the subscription doe snot change for the sharee

We could use the subscription probe to check the sharee is still subscribed to the mailbox? (just another assertion for certainty, not exclusive your await way).

Reply:
Hmm i get it... My point is if we use the awaitUntil to check tha the subscription changes, it's because it's reactive and does not happens right away (specially in distributed). So my point is how do we make sure the change never happens for the sharee? It could be we check the change happens on Bob... but maybe a change happens on Alice later (which we do not want) and we might miss it... 

But maybe I'm thinking too much here :)

Reply:
I wait a second between the two with thread sleep and I check we don't have mailbox/changes for the sharee when owner changes his subscription to the mailbox... should be enough i hope :)

i will try to do something similar in MailboxSetContract and check the subscription does not change for sharees if owner change it for example

Reply:
Actually no need, there is already a test for that in that contract: MailboxSetMethodContract.updateShouldNotAffectSubscriptionOfOthers

Reply:
For another user (Andre)
After Bob receives the changed `"updated": ["BobMailbox1"]`
We can change AndreMailbox1, and AwaitUntil the test for ensuring Andre Mailbox changes only AndreMailbox1 (does not contain BobMailbox1)

It not sure 100%, but it will be more trusty

Reply:
Good proposition actually, I will look into that thanks :)